### PR TITLE
new flag: onlyReconcileRules

### PR DESCRIPTION
**Pull request by evenh**
When enabled the operator will only reconcile LokiRule's into the ConfigMap. It will skip updating the DaemonSet hash and will avoid restarts of Loki.

This is useful in environments where Loki reloads the mounted ConfigMap by itself at a fixed interval. This avoids downtime when using PVs.

### CODE REVIEW

**Comment by evenh** [comment]
Would it be possible to give this pull request a review? 🤞🏻

@BrunoTarijon

**Review by BrunoTarijon — COMMENTED** [review]
Nice, the hot reload is working by default with the oficial loki chart with the simple scalable deploy, I think the default value should be true.

Just one thing, doing this the controller do not mount the configmap, this make sense? It should at least verify if the configmap is mounted (Yes - do nothing) (No - mount the configmap).

**Comment by BrunoTarijon** [comment]
Ok, this flag should prevent the operator from changing the Loki.

Can you do the test? When the flag is true the controller should not mount the configmap.

**Comment by evenh** [comment]
I've rebased against main, modified the tests and bumped the versions.

File: deploy/helm/charts/loki-rule-operator/Chart.yaml [hunk 1]
```diff
@@ -15,10 +15,10 @@ type: application
...
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: "0.7.4"
+version: "0.7.5"
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "0.6.2"
```
Comment (BrunoTarijon) [inline deploy/helm/charts/loki-rule-operator/Chart.yaml:24]: The image.tag is [set in the values.yaml](https://github.com/quero-edu/loki-rule-operator/blob/main/deploy/helm/charts/loki-rule-operator/values.yaml#L4). Can you remove it from the values? We will use the appVersion to set the default image tag for now on.

That done, I will merge this PR

**Review by BrunoTarijon — APPROVED** [review]

**Pull request merged**